qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -259,9 +258,9 @@ def expectation_value(self, oper, qargs=None):
             pauli_z = pauli_z ^ stab.Z[p]
 
         if phase % 4 != 0:
-            return -1
+            return -pauli_phase
```

Comment:
Is this test correct?  It seems to partition the symplectic phase such that 1 maps to 1, but -1, i and -i all map to -1.  I don't know enough about stabiliser states to know for sure if that's right, so I just wanted to check - it _feels_ like 1 and -1 should behave the same, as should i and -i, but that's entirely guesswork.

Reply:
This code is correct. Note that in line 223 we defined:
`pauli_phase = (-1j) ** oper.phase if oper.phase else 1`
So, `pauli_phase` and `-pauli_phase` can get all the values 1,-1,i,-i.

Reply:
Sorry, I wasn't clear here - I mean a `phase` representing 1 keeps `pauli_phase` the same, but a `phase` of -1, i or -i negates it. My (probably incorrect) mathematical intuition about expectation values thought that the natural link would be `phase` squared, where a `phase` of -1 would also leave `pauli_phase` unchanged.

Reply:
I don't think it's easy to explain how the code works, but note that beforehand it was correct for unsigned Paulis.
Now, if you multiply the oper by a constant, then the expectation value will also be multiplied by the same constant, and this is the main change in the current code

Reply:
It doesn't seem to be possible for this value to ever be odd - I ran a bunch of tests generating random stabiliser states and taking the expectation value with every possible Pauli for that number of qubits, and I could never get an odd number in `phase` after the loop.  That at least solves the problem I was having of weird intuition - if `phase` can't ever be 1 or 3 then there's no such a problem.

Is that expected from the formalism?  If so, perhaps there's a shortcut somewhere, because there might be a factor of two we can reduce.  At the very least, it might be clearer for the future if we write it as `if (phase // 2) % 2 == 1:` (or add a comment here) explaining why there's only two possible values.

Reply:
yes, phase can be only 0 or 2.

Reply:
Ahhhh, this is where all my problem was - that's the thing I was missing.  I've pushed up a single comment on that, and I'll merge the PR now.